cron/cron.go

```diff
@@ -143,6 +143,11 @@ func startFunc(wg *sync.WaitGroup, exitCtx context.Context, logger *logrus.Entry
 		var cronIteration uint64
 		nextRun := time.Now()
 
+		if expression == crontab.StartExpr {
+			fn(nextRun, logger)
+			return
+		}
```

Comment:
I'd rather handle this purely within the expression interface. Right now, this feels a little hacky, and as a result it means start jobs will not get to finish if you CTRL+C (`jobWg.Wait()` won't wait for anything and `wg.Done()` will be called immediately). We also, get incomplete logging.

To do this cleanly, I think we need 2 changes to the interface for `expression.Next()`:

- Give it a counter of how many times we have gone through this loop already (we ahve this variable in `cronIteration` already).
- Return a pointer to a time, then we can check for null and break out of this loop if null.